Week3/app/src/main/java/com/florencenjeri/businesscard/MainActivity.kt

```diff
@@ -1,11 +1,88 @@
 package com.florencenjeri.businesscard
 
-import androidx.appcompat.app.AppCompatActivity
+import android.content.Context
+import android.content.Intent
+import android.content.SharedPreferences
 import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.animation.AnimationUtils
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.app.AppCompatDelegate
+import com.raywenderlich.airlock.Constants
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
+
+    lateinit var prefs: SharedPreferences
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        prefs = this.getSharedPreferences(Constants.PREFS_MODE, Context.MODE_PRIVATE)
+        when (prefs.getInt(Constants.MODE_KEY, 0)) {
```

Comment:
Maybe pull this, and the listener code functions somewhere below, to make this behavior reusable!